Support create/delete veth interface

cathay4t opened the pull request:
Example yaml for `npc set`:

```yml
---
ifaces:
  - name: veth1
    type: veth
    veth:
      peer: veth1.ep
    ipv4:
      addresses:
        - address: "192.0.2.1"
          prefix_len: 24
    ipv6:
      addresses:
        - address: "2001:db8:a::9"
          prefix_len: 64
```

To delete a interface, you may:

```yml
---
ifaces:
  - name: veth1
    state: absent
```

The creating of 1000 veth interface(including another 1000 veth peer)
takes 5.120 seconds.

Integration tests for creating and delete veth have been included.

cathay4t commented:
@ffmancera Patch ready for review.

ffmancera commented:
Big patch, taking a lot. I will let you know tomorrow my thoughts. Thanks!

cathay4t commented:
The performance is bad(8m33.713s) when trying to create 1000 veth pairs. Will improve it.

qinqon commented on src/lib/ip.rs:
```diff
@@ -61,6 +62,37 @@ pub struct IpConf {
     pub addresses: Vec<IpAddrConf>,
 }
 
+impl From<&Ipv4Info> for IpConf {
+    fn from(info: &Ipv4Info) -> Self {
+        let mut addrs = Vec::new();
+        for addr_info in &info.addresses {
+            if addr_info.valid_lft == "forever" {
+                addrs.push(IpAddrConf {
+                    address: addr_info.address.clone(),
+                    prefix_len: addr_info.prefix_len,
+                });
+            }
+        }
+        Self { addresses: addrs }
+    }
+}
+
+impl From<&Ipv6Info> for IpConf {
+    fn from(info: &Ipv6Info) -> Self {
+        let mut addrs = Vec::new();
+        for addr_info in &info.addresses {
+            if addr_info.valid_lft == "forever" {
+                addrs.push(IpAddrConf {
+                    address: addr_info.address.clone(),
+                    prefix_len: addr_info.prefix_len,
+                });
+            }
+        }
+        Self { addresses: addrs }
+    }
+}
+
```
Can we make a ipv4/ipv6 generic function for this ?

cathay4t replied:
They are taking different type of input. Maybe [`Box<dyn Trait>`][1] might help, but I don't like to include complexity which is unfriendly to me and other new rust learner. If you have suggestions, please kindly let me know.


[1]: https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/dyn-trait-for-trait-objects.html

qinqon commented on src/lib/ip.rs:
```diff
@@ -81,141 +113,254 @@ impl IpConf {
+                    prefix_len: cur_addr.prefix_len,
+                });
+            }
+            let has_ipv6_link_local_in_desire = if ip_family == IpFamily::Ipv4 {
```
This does not have to be `ip_family == IpFamily::Ipv4`

cathay4t replied:
It saves some cpu time on iteration of `is_ipv6_unicast_link_local_full` over ipv4 addresses.

qinqon reviewed (changes requested):
First pass, a pair of small things/questions

cathay4t commented:
@ffmancera Patch ready for review.

cathay4t committed:
apply veth: support creating and deleting veth

Example yaml for `npc set`:

```yml
---
ifaces:
  - name: veth1
    type: veth
    veth:
      peer: veth1.ep
    ipv4:
      addresses:
        - address: "192.0.2.1"
          prefix_len: 24
    ipv6:
      addresses:
        - address: "2001:db8:a::9"
          prefix_len: 64
```

To delete a interface, you may:

```yml
---
ifaces:
  - name: veth1
    state: absent
```

The creating of 1000 veth interface(including another 1000 veth peer)
takes 5.120 seconds.

Integration tests for creating and delete veth have been included.
Example YAML files included for creating and deleting veth.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Let's merge this and get it going. Other PR is waiting this.
